src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -0,0 +1,182 @@
+    time_end = time_start + timedelta(minutes=5)
+
+    # ... wait for 5 minutes until the buffer is filled
+    await asyncio.sleep(5)
+
+    # return an numpy array from the window
+    a = window[time_start:time_end]
+    # and use it to for example calculate the mean
+    mean = a.mean()
+    '''
+
+    **Example2** (create a polars data frame from a `MovingWindow`):
+
+    ```
+    import polars as pl
+
+    # create a window that stores two days of data
+    # starting at 1.1.23 with samplerate=1
+    window = MovingWindow(size = (60 * 60 * 24 * 2), sample_receiver)
+
+    # wait for one full day until the buffer is filled
+    asyncio.sleep(60*60*24)
+
+    # create a polars series with one full day of data
+    time_start = datetime(2023, 1, 1)
+    time_end = datetime(2023, 1, 2)
+    s = pl.Series("Jan_1", mv[time_start:time_end])
+    ```
+    """
+
+    def __init__(
+        self,
+        size: int,
+        resampled_data_recv: Receiver[Sample],
+        sampling_period: timedelta,
+        window_alignment: datetime = datetime(1, 1, 1),
+    ) -> None:
+        """
+        Initialize the MovingWindow.
+
+        This method creates the underlying ringbuffer and starts a
+        new task that updates the ringbuffer with new incoming samples.
+        The task stops running only if the channel receiver is closed.
+
+        Args:
+            size: The number of elements that are stored.
+            resampled_data_recv: A receiver that delivers samples with a
+                given sampling period.
+            sampling_period: The sampling period.
+            window_alignment: A datatime object that defines a point in time to which
+                the window is aligned to modulo window size.
+                (default is midnight 01.01.01)
+
+        Raises:
+            asyncio.CancelledError: when the task gets cancelled.
+        """
+        self._resampled_data_recv = resampled_data_recv
+        self._buffer = OrderedRingBuffer(
+            np.empty(shape=size, dtype=float),
+            sampling_period=sampling_period,
+            time_index_alignment=window_alignment,
+        )
+        self._copy_buffer = False
+        try:
+            _t = asyncio.create_task(self._run_impl())
+            # make the linter happy
+            assert isinstance(_t, asyncio.Task)
+        except asyncio.CancelledError:
+            log.debug("Cancelling MovingWindow task: %s", __name__)
+            raise
+
+    async def _run_impl(self) -> None:
+        """Awaits samples from the receiver and updates the underlying ringbuffer."""
+        async for sample in self._resampled_data_recv:
+            log.debug("Received new sample: %s", sample)
+            log.debug("Updating ring buffer with latest sample")
+
+            # missing values coming from the logical meter are marked as NaN
+            missing = False
+            assert isinstance(sample.value, float)
+            if isnan(sample.value):
```

Comment:
what about `if isinstance(sample.value, float) and isnan(sample.value):` and remove the assert above?

Reply:
No I would prefer the above, because the check if the value is float is again a linter thing. But this logic is moved to the ringbuffer anyways and will be deleted with the next rebase.

Reply:
oh I see, then it is fine as it is...